src/routes/api/v1/dataset.router.js

```diff
@@ -558,6 +573,24 @@ const authorizationMiddleware = async (ctx, next) => {
     await next(); // SUPERADMIN is included here
 };
 
+const deleteResourceAuthorizationMiddleware = async (ctx, next) => {
+    logger.info(`[DatasetRouter] Checking authorization`);
+    const user = DatasetRouter.getUser(ctx);
+    const userFromParam = ctx.params.userId;
+
+    if (user.id === 'microservice' || user.role === 'ADMIN') {
+        await next();
+        return;
+    }
+
+    if (userFromParam !== user.id) {
+            ctx.throw(403, 'Forbidden');
```

Comment:
why the evil formatting? get VSCode or whatever you are using to auto-format on save.